### PR TITLE
[Training] Add VAE precompute

### PR DESCRIPTION
**Pull request by sang-neto**
Also in this PR:
- Add gradient clipping
- Add training code to load npz from vae precomputed

Test code:
```py
import numpy as np


data = np.load("file.npz")


import torch
from diffusers import AutoencoderOobleck

# Load VAE model
vae = AutoencoderOobleck.from_pretrained(
    "stabilityai/stable-audio-open-1.0", subfolder="vae"
)
vae.eval()
vae.to("cuda:0")

# Convert numpy array to torch tensor and move to GPU
latent = torch.from_numpy(data['tensor']).unsqueeze(0).cuda()


# Decode latent back to audio waveform
with torch.no_grad():
    wave = vae.decode(latent).sample.cpu()[0]

# Wave output will be in shape (channels, samples)
# Transpose to (samples, channels) if needed for audio playback
# wave = wave.transpose(0, 1)

print("Decoded audio shape:", wave.shape)
print("Sample rate:", vae.config.sampling_rate)

# Optional: Save as wav file
import torchaudio
torchaudio.save('decoded_audio.wav', wave, vae.config.sampling_rate)

```

### CODE REVIEW

**Comment by Saltb0xApps** [comment]
we also need to modify train.py to not load the VAE during training, and update the pipeline to read npz instead of the dataset
